security/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+by Kubernetes. Metal3 provides comprehensive set of components for baremetal
+host management with Kubernetes so that a user can enroll user's baremetal
+machines, provision operating system images, deploy Kubernetes clusters on them
+and maintain the lifecycle of the kubernetes cluster, baremetal server and
+applications all through kubernetes native APIs.
+
+### Background
+
+Metal3 is an open-source technology which enables users to provision and
+maintain a baremetal server's lifecycle using Kubernetes native APIs. There are
+a number of great open source tools for bare metal host provisioning, including
+Ironic. Metal3 aims to build on these technologies to provide a Kubernetes
+native API for managing bare metal hosts via a provisioning stack that is also
+running on Kubernetes. We believe that Kubernetes Native Infrastructure, or
+managing your infrastructure just like your applications, is a powerful next
+step in the evolution of infrastructure management.
+
+The Metal3 project is also building integration with the Kubernetes Cluster API
+project, allowing Metal3 to be used as an infrastructure backend for Machine
+objects from the Cluster API. These components integrate seamlessly to leverage
+the Kubernetes ecosystem and automate the provisioning and management of
+bare-metal infrastructure.
+
+This is paired with one of the components from the OpenStack ecosystem, Ironic
+for booting and installing machines. Metal³ handles the installation of Ironic
+as a standalone component (there’s no need to bring along the rest of
+OpenStack). Ironic is supported by a mature community of hardware vendors and
+supports a wide range of bare metal management protocols which are continuously
+tested on a variety of hardware. Backed by Ironic, Metal³ can provision
+machines, no matter the brand of hardware.
+
+### Actors
+
+Metal3 as a project bundles the following actors together:
+
+#### Cluster API provider Metal3 (CAPM3)
+
+The Cluster API brings declarative, Kubernetes-style APIs to cluster creation,
+configuration and management. The API itself is shared across multiple cloud
+providers. Cluster API Provider Metal3(CAPM3) is one such provider for Cluster
+API which enables users to deploy a Cluster API based cluster on top of bare
+metal infrastructure using Metal3. In one hand it acts as a plugin for Custer
+API and brings in the flexibility of simplified kubernetes cluster management,
+and on the other hand it interfaces with Baremetal operator (a controller which
+manages Baremetal Host API) to manage baremetal servers lifecycle through simple
+kubernetes native APIs.
+
+#### IP Address Manager (IPAM)
+
+The IP Address Manager (IPAM) is actor in Metal3 which manages static IP
+allocations for baremetal hosts. IPAM handles allocations of IP subnet according
+to the requests without handling any use of those addresses. The IPAM adds the
+flexibility by providing the address right before provisioning the node. It can
```

Comment:
```suggestion
The IP Address Manager (IPAM) is an actor in Metal3 which manages static IP
allocations for baremetal hosts. IPAM handles allocations of IPs from subnets according
to the requests without handling any use of those addresses. IPAM adds the
flexibility by providing the address right before provisioning the node. It can
```

> IPAM adds the flexibility by providing the address right before provisioning the node.

I'm not understanding this sentence. How is it providing flexibility?
Maybe we can change it to a continuation on the previous. Something like

```
IPAM simply keeps track of IP pools and allocations. It is then up to the consumers to act on the information.
```

Reply:
@kashifest I think this is from you, what do you think?

Reply:
I am ok with @lentzi90's suggestion

Reply:
Discussed with Kashif offline, we go with your suggestion.

Fixed now.